web/modules/custom/error_page/src/ErrorPageErrorHandler.php

```diff
@@ -0,0 +1,281 @@
+      _drupal_error_header($error['@message'], $error['%type'], $error['%function'], $error['%file'], $error['%line']);
+    }
+
+    $response = new Response();
+
+    // Only call the logger if there is a logger factory available. This can
+    // occur if there is an error while rebuilding the container or during the
+    // installer.
+    if (\Drupal::hasService('logger.factory')) {
+      try {
+        // Provide the PHP backtrace to logger implementations.
+        \Drupal::logger('php')->log($error['severity_level'], '%type: @message in %function (line %line of %file) [@uuid] @backtrace_string', $error + ['backtrace' => $backtrace]);
+      }
+      catch (\Exception $e) {
+        // We can't log, for example because the database connection is not
+        // available. At least try to log to PHP error log.
+        error_log(strtr('Failed to log error: %type: @message in %function (line %line of %file) [@uuid]. @backtrace_string', $error));
+      }
+    }
+
+    // Log fatal errors, so developers can find and debug them.
+    if ($fatal) {
+      if ($uuid) {
+        error_log(sprintf('%s: %s in %s on line %d [%s] %s', $error['%type'], $error['@message'], $error['%file'], $error['%line'], $error['@uuid'], $error['@backtrace_string']));
+      }
+      else {
+        error_log(sprintf('%s: %s in %s on line %d %s', $error['%type'], $error['@message'], $error['%file'], $error['%line'], $error['@backtrace_string']));
+      }
+    }
+
+    if (PHP_SAPI === 'cli') {
+      if ($fatal) {
+        // When called from CLI, simply output a plain text message. Should not
+        // translate the string to avoid errors producing more errors.
+        $response->setContent(html_entity_decode(strip_tags(new FormattableMarkup('%type: @message in %function (line %line of %file) [@uuid].', $error))) . "\n");
+        $response->send();
+        exit;
+      }
+    }
+
+    if (\Drupal::hasRequest() && \Drupal::request()->isXmlHttpRequest()) {
+      if ($fatal) {
+        if (error_displayable($error)) {
+          // When called from JavaScript, simply output the error message.
+          // Should not translate the string to avoid errors producing more
+          // errors.
+          $response->setContent(new FormattableMarkup('%type: @message in %function (line %line of %file) [@uuid].', $error));
+          $response->send();
+        }
+        exit;
+      }
+    }
+    else {
+      $class = 'error';
+      // If error type is 'User notice' then treat it as debug information
+      // instead of an error message.
+      // @see debug()
+      if ($error['%type'] == 'User notice') {
+        $error['%type'] = 'Debug';
+        $class = 'status';
+      }
+
+      $message = $error_report = '';
+
+      // Generate an error report if the error reporting level allows it.
+      if (error_displayable($error)) {
+        if (_drupal_get_error_level() === ERROR_REPORTING_DISPLAY_VERBOSE) {
```

Comment:
Compared to core default error logger, we are ignoring some error levels: `ERROR_REPORTING_DISPLAY_ALL` and `ERROR_REPORTING_DISPLAY_SOME`. If possible we should honour also those levels.